woocommerce/woocommerce-ios
# [Mobile payments] Add networking - plugins

Partially addresses #3839

Why are we doing this?
- This will allow us to, as part of determining if a site supports card present payments, get a list of active and inactive plugins.

Note:
- This is in support of card-present-payments, but since it doesn't directly depend on card-present-payments, I've elected to submit this against the `develop` branch, and not `feature/stripe-terminal-sdk-integration`

What it does:
- Adds networking layer support for fetching the plugins on a site, both active and inactive

How fast is it?
- Not very. On low-end shared hosting, the plugins endpoint typically takes 4-5 seconds to respond.

To test:
- Ensure SitePluginsMapperTests pass
- Ensure SitePluginsRemoteTests pass
- If you'd like to test with a real request against a real site, a quick test would be to modify `Yosemite/Yosemite/Stores/SettingStore.swift` as follows:

```diff
diff --git a/Yosemite/Yosemite/Stores/SettingStore.swift b/Yosemite/Yosemite/Stores/SettingStore.swift
index cb5ac8b18..7cd7726d3 100644
--- a/Yosemite/Yosemite/Stores/SettingStore.swift
+++ b/Yosemite/Yosemite/Stores/SettingStore.swift
@@ -8,6 +8,7 @@ import Storage
 public class SettingStore: Store {
     private let siteSettingsRemote: SiteSettingsRemote
     private let siteAPIRemote: SiteAPIRemote
+    private let sitePluginsRemote: SitePluginsRemote
 
     private lazy var sharedDerivedStorage: StorageType = {
         return storageManager.newDerivedStorage()
@@ -16,6 +17,7 @@ public class SettingStore: Store {
     public override init(dispatcher: Dispatcher, storageManager: StorageManagerType, network: Network) {
         self.siteSettingsRemote = SiteSettingsRemote(network: network)
         self.siteAPIRemote = SiteAPIRemote(network: network)
+        self.sitePluginsRemote = SitePluginsRemote(network: network)
         super.init(dispatcher: dispatcher, storageManager: storageManager, network: network)
     }
 
@@ -62,6 +64,9 @@ private extension SettingStore {
                 onCompletion(nil)
             }
         }
+        sitePluginsRemote.loadPlugins(for: siteID) { [weak self] (plugins, error) in
+
+        }
     }
```
and then set a breakpoint inside that completion. When the app launches and fetches settings, it will also fetch plugins and show them in the watch window.

Update release notes:
